### PR TITLE
Fix Pressing Escape doesn't exit freelook in the 3D editor

### LINKED ISSUE #95141: Pressing Escape doesn't exit freelook in the 3D editor
### Tested versions

- Reproducible in: 3.5.2.stable, 4.2.2.stable, 4.3.rc e343dbbcc

### System information

Godot v4.3.beta (e343dbbcc) - Fedora Linux 40 (KDE Plasma) - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 (nvidia; 555.58.02) - 13th Gen Intel(R) Core(TM) i9-13900K (32 Threads)

### Issue description

Pressing Escape doesn't exit freelook in the 3D editor, even though the editor has code for it:

https://github.com/godotengine/godot/blob/3978628c6cc1227250fc6ed45c8d854d24c30c30/editor/plugins/node_3d_editor_plugin.cpp#L2409-L2411

When you press <kbd>Escape</kbd>, something else happens instead, such as having the current node be deselected in the inspector.

### Steps to reproduce

- Enable freelook in the 3D editor by pressing <kbd>Shift + F</kbd> or holding down the right mouse button.
  - Pressing <kbd>Escape</kbd> should be able to disable freelook (and not deselect nodes the first time you do it) in both cases, but it doesn't.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by jsjtxietian**
Hacky way to fix https://github.com/godotengine/godot/issues/95141

### CODE REVIEW

File: editor/plugins/node_3d_editor_plugin.cpp [hunk 1]
```diff
@@ -2252,6 +2252,11 @@ void Node3DEditorViewport::_sinput(const Ref<InputEvent> &p_event) {
 		}
 
 		if (_edit.mode == TRANSFORM_NONE) {
+			if (!_edit.gizmo.is_valid() && is_freelook_active()) {
```
Comment (AThousandShips) [inline editor/plugins/node_3d_editor_plugin.cpp:2255]:
```suggestion
			if (_edit.gizmo.is_null() && is_freelook_active()) {
```

Comment (Calinou) [reply editor/plugins/node_3d_editor_plugin.cpp:2255]: This is missing a check for `k->get_keycode() == Key::ESCAPE` too. Right now, pressing *any* key will exit freelook.

Comment (jsjtxietian) [reply editor/plugins/node_3d_editor_plugin.cpp:2255]: Thanks!

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.3.1.

**Pull request merged**